src/test/kotlin/study/StringCalculator.kt

```diff
@@ -0,0 +1,72 @@
+package study
+
+class StringCalculator {
+    fun calculate(arithmeticExpression: String): Float {
+        if (arithmeticExpression.isBlank()) {
+            throw IllegalArgumentException("문자열이 비어있습니다.")
+        }
+
+        val splitedExpression = splitByBlank(arithmeticExpression)
+        val numbers = splitedExpression.filter { isNumber(it) }
+        val arithmeticSymbols = splitedExpression.filter { isOperator(it) }
+
+        validate(splitedExpression, numbers, arithmeticSymbols)
+
+        val result = numbers.map { it.toFloat() }
+            .toTypedArray()
+            .reduceIndexed { index, acc, s ->
+                doCalculate(acc, s, arithmeticSymbols[index - 1])
+            }
+
+        println(result)
+        return result
+    }
+
+    private fun validate(splitedExpression: List<String>, numbers: List<String>, arithmeticSymbols: List<String>) {
+        if (splitedExpression.size != numbers.size + arithmeticSymbols.size) {
+            throw IllegalArgumentException("숫자와 기호외의 문자가 포함되었습니다.")
+        }
```

Comment:
훌륭한 조건이네요! 아래와 같은 코드는 어떨까요? 조금 더 명확할 수 있어 보여요!

```suggestion
        if (!splitedExpression.all { it in listOf("0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "+", "-", "*", "/") } ) {
            throw IllegalArgumentException("숫자와 기호외의 문자가 포함되었습니다.")
        }
```